docs/docs/render-callbacks.md

```diff
@@ -0,0 +1,56 @@
+---
+id: render-callbacks
+title: Render Callbacks
+permalink: docs/render-callbacks.html
+---
+
+Render callbacks are an advanced technique for certain types of asynchronous rendering. A render callback is a function that is passed as a prop to a React element, and is called at some point in the future. The render callback is typically passed additional data that isn't available at the time the element is rendered:
+
+```js
+render() {
+  return (
+    <FetchData
```

Comment:
I found this hard to understand initially because `<FetchData/>` isn't a thing. Is there any example for where you'd use render callbacks that is simple enough that you could actually include real working code? Bonus points for codepen. Maybe something like, a render callback to render the ith thing in a list?